lib/cosmos/packet_logs/packet_log_reader.rb

```diff
@@ -152,37 +98,28 @@ def close
     def read(identify_and_define = true)
       # Read entry length
       length = @file.read(4)
-      return nil if !length or length.length == 0
+      return nil if !length or length.length <= 0
       length = length.unpack('N')[0]
-      #STDOUT.puts "length = #{length}"
-
       entry = @file.read(length)
       flags = entry[0..1].unpack('n')[0]
-      #STDOUT.printf("flags = 0x%0X\n", flags)
 
       cmd_or_tlm = :TLM
       cmd_or_tlm = :CMD if flags & COSMOS5_CMD_FLAG_MASK == COSMOS5_CMD_FLAG_MASK
-      #STDOUT.puts "cmd_or_tlm = #{cmd_or_tlm}"
       stored = false
       stored = true if flags & COSMOS5_STORED_FLAG_MASK == COSMOS5_STORED_FLAG_MASK
-      #STDOUT.puts "stored = #{stored}"
       id = false
       id = true if flags & COSMOS5_ID_FLAG_MASK == COSMOS5_ID_FLAG_MASK
-      #STDOUT.puts "id = #{id}"
 
       if flags & COSMOS5_JSON_PACKET_ENTRY_TYPE_MASK == COSMOS5_JSON_PACKET_ENTRY_TYPE_MASK
-        #STDOUT.puts "JSON_PACKET"
         packet_index, time_nsec_since_epoch = entry[2..11].unpack('nQ>')
         json_data = entry[12..-1]
-        #STDOUT.puts json_data.inspect
-        cmd_or_tlm, target_name, packet_name, id = @packets[packet_index]
+        target_name, packet_name, id = @packets[packet_index]
```

Comment:
cmd_or_tlm is determined above by the flags. This was overwriting the value with TLM all the time which was breaking the spec.

Reply:
The packet at each packet index is specifically either a command packet or a telemetry packet.   The packet index lookup should return cmd_or_tlm, and if you want, could do a check against the entry cmd_or_tlm to make sure they match.

Reply:
What's the point of storing this when we look it up at the beginning of the method using the flags? This overwriting of cmd_or_tlm was the source the bug when I was testing.